Closes #2788 fix_links: use vignette("vignette name") in function documentation

bundfussr opened the pull request:
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/cran-release/CONTRIBUTING.html#detailed-development-process) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `main` branch until you have checked off each task.

- [ ] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [ ] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [ ] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [ ] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#deprecation)?
- [ ] Review the [Cheat Sheet](https://github.com/pharmaverse/admiral/blob/main/inst/cheatsheet/admiral_cheatsheet.pdf). Make any required updates to it by editing the file `inst/cheatsheet/admiral_cheatsheet.pptx` and re-upload a PDF and a PNG version of it to the same folder. (The PNG version can be created by taking a screenshot of the PDF version.)
- [ ] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [ ] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [ ] Address any updates needed for vignettes and/or templates
- [ ] Update `NEWS.md` under the header `# admiral (development version)` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers). A Developer Notes section is available in `NEWS.md` for tracking developer-facing issues.  
- [ ] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/cran-release/reference/index.html)" page. 
- [ ] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [ ] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [ ] Link the issue in the Development Section on the right hand side.
- [ ] Address all merge conflicts and resolve appropriately
- [ ] Pat yourself on the back for a job well done! Much love to your accomplishment!

bms63 reviewed (commented):
Since this is a patch to CRAN - do you mind making a new 1.3.1 news entry for this and updating DESCRIPTION in this PR?

I think I can send to CRAN after that??

bms63 commented:
oh we have that too ... i forgot about that one.  yeah lets just release it as well then we don't have to fuss with the tag stuff for 1.3.0.  it is such a minor update

bundfussr commented:
> oh we have that too ... i forgot about that one. yeah lets just release it as well then we don't have to fuss with the tag stuff for 1.3.0. it is such a minor update

Version number and NEWS is updated.

bms63 commented:
I haev submitted, but waiting to hear back from CRAN before we merge into `main`

We might need to fix this issue:
https://[win-builder.r-project.org/incoming_pretest/admiral_1.3.1_20250728_180345/Windows/00check.log](https://win-builder.r-project.org/incoming_pretest/admiral_1.3.1_20250728_180345/Windows/00check.log)
<img width="1307" height="606" alt="image" src="https://github.com/user-attachments/assets/2223dd7c-86c3-4c5e-b88d-c655d8d4388b" />

bundfussr commented:
> I haev submitted, but waiting to hear back from CRAN before we merge into `main`
> 
> We might need to fix this issue: https://[win-builder.r-project.org/incoming_pretest/admiral_1.3.1_20250728_180345/Windows/00check.log](https://win-builder.r-project.org/incoming_pretest/admiral_1.3.1_20250728_180345/Windows/00check.log) <img alt="image" width="1307" height="606" src="https://private-user-images.githubusercontent.com/10111024/471617551-2223dd7c-86c3-4c5e-b88d-c655d8d4388b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTM3NzIyMTAsIm5iZiI6MTc1Mzc3MTkxMCwicGF0aCI6Ii8xMDExMTAyNC80NzE2MTc1NTEtMjIyM2RkN2MtODZjMy00YzVlLWI4OGQtYzY1NWQ4ZDQzODhiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA3MjklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNzI5VDA2NTE1MFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTU0NGQ0NzM2ZDYzOTVjMDRiMDcwYTA5YjVlOGM2ZTAzZTk0OGVhMThlZjAxOGM4YWY2YjcwZGNiMGE5NGU1OTYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.OkNu18jWY15os4CjFY4zIsqcWr4dF0bWl9U2buDtCZg">

We definitely need to remove "Remotes:". Sorry I missed that.

I have removed it now and updated the CTCAE links. I've also updated the minimum dplyr version (the `relationship` argument was introduced in dplyr 1.1.1; most likely this causes the issue reported in https://github.com/pharmaverse/admiral/issues/2793).

bms63 commented:
@bundfussr is this dplyr 1.1.1 issue news worthy?  feels like we should document it?
